scanner/decisions/0005-scannerv4-read-cvss-data.md

```diff
@@ -0,0 +1,36 @@
+
+We've chosen to base our API requests on CVSS V3 severity to ensure all data integrated into Claircore possesses valid CVSS V3 metrics. Consequently, retrieving data without CVSS V3 information becomes irrelevant.
+
+The CVSS Updater in Central will be modified to download zip file (generated by CI mentioned above) in the Google bucket. This eliminates the need for JSON parsing in the updater. ScannerV4 will pull the zip bundle from Central and populate the Matcher DB with the enrichment data.
+
+The CVSS Updater in Central will attempt to download the data bundle from the Google bucket a maximum of five times. Should all five attempts be unsuccessful, the updater will pause its efforts and initiate a fresh download attempt after a 4-hour interval. After five unsuccessful download attempts, a failure notification will be swiftly sent to the Scanner on-call Slack channel.
+
+## Consequences
+
+One drawback of the NVD CVE API is its rate limit: only 5 requests are allowed per 30 seconds without an API Key. This restriction necessitates that our workflow fetch data at a slower pace. Additionally, transitioning from the NVD JSON feed to the NVD CVE API brings in pagination limitations. For example, when fetching data with a "low" severity, the API provides a maximum of 2000 CVEs per response. This means multiple requests are required to retrieve the complete dataset for "low" severity. The result is an increased number of smaller JSON files. Consequently, we need to compress all JSON files into a single zip archive.
+
+Changing from the NVD JSON feed to the NVD CVE API also give us the conveniences of no need to filter out data without valid CVSS V3 metrics, as mentioned above. 
+
+Following this modification, Central will no longer have a dependency on Claircore. The Claircore enricher will be used in Scanner V4, which adopts Claircore as its primary scanning engine by design.
+
+Per the NVD's documentation (https://nvd.nist.gov/developers/vulnerabilities), the NVD CVE API omits CVSS v3 vector strings with a 'NONE' severity. This ensures we only obtain data bearing valid CVSS V3 metrics.
+
+If the GH workflow (CI) encounters issues accessing or downloading portions of the CVSS data—likely due to NVD API unavailability or rate limiting—it should retry the request after a 6-second delay, capping at three total attempts. If the same HTTP request fails three times consecutively, that specific data section will be skipped. Central retains the data bundle from the prior update. In cases where the current update has data missing but the previous one doesn't (as determined by comparing the sizes of the two data bundles), the earlier data bundle will be employed to support Scanner v4.
```

Comment:
That's a decision, not a consequence. Move to the "Decision" section. If you want to, mention what this decision will provide to us (in this case, stale data will be served to Scanner until the latest vulns are pulled from the bucket).